exam_1/task2/main.c

```diff
@@ -0,0 +1,88 @@
+#include <stdio.h>
+#include <string.h>
+
+typedef struct Product
+{
+    int amount;
+    char name[100];
+} Product;
+
+void bubbleSort(Product *prodcts, int size, char *iterator)
+{
+    for (int i = 0; i < size - 1; i++)
+    {
+        for (int j = (size - 1); j > i; j--)
+        {
+            if (prodcts[j - 1].amount > prodcts[j].amount && strcmp(iterator, "amount") == 0 || strcmp(prodcts[j - 1].name, prodcts[j].name) < 0 && strcmp(iterator, "name") == 0)
+            {
+                Product temp = prodcts[j - 1];
+                prodcts[j - 1] = prodcts[j];
+                prodcts[j] = temp;
+            }
+        }
+    }
+}
+
+int main()
+{
+    Product products[1000] = {0};
+    FILE *file = fopen("storage.txt", "r");
+    if (file == NULL)
+    {
+        printf("File were not open");
+        return -1;
+    }
+    int size = 0;
+
+    int products50 = 0;
+    int products25 = 0;
+
+    printf("Enter amount of products with 50%% discount: ");
+    scanf("%d", &products50);
+    if (products50 < 0)
+    {
+        printf("Wrong input: anount of products with 50%% discount should be positive or should equal to zero\n");
+        return 0;
+    }
+
+    printf("Enter amount of products with 25%% discount: ");
+    scanf("%d", &products25);
+    if (products25 < 0)
+    {
+        printf("Wrong input: anount of products with 25%% discount should be positive or should equal to zero\n");
+        return 0;
+    }
+    printf("\n");
+    char c = 0;
+    for (; size < 1000; size++)
+    {
+        if (fscanf(file, "%s %c %d", products[size].name, &c, &products[size].amount) == EOF)
```

Comment:
```suggestion
        if (fscanf(file, "%s - %d", products[size].name, &products[size].amount) == EOF)
```
Нет проверки, что название и количество разделены дефисом (-0.25)
А если файл имеет некорректную структуру? Тогда `fscanf` вернет значение от 0 до 2. Основываясь на этом, нужно было определить, что файл некорректен. (-0.5)